Libplanet.Extensions.Cocona/Commands/TxCommand.cs

```diff
@@ -0,0 +1,153 @@
+        )
+        {
+            if (unsigned && ignoreSignature)
+            {
+                throw new CommandExitedException(
+                    "The -u/--unsigned and -i/--ignore-signature options are mutually exclusive.",
+                    -1
+                );
+            }
+
+            byte[] bytes;
+            string sourceName;
+            if (file == "-")
+            {
+                sourceName = "stdin";
+                using var stdin = Console.OpenStandardInput();
+                using var buffer = new MemoryStream();
+                stdin.CopyTo(buffer);
+                bytes = buffer.GetBuffer();
+                Array.Resize(ref bytes, (int)buffer.Length);
+            }
+            else
+            {
+                sourceName = $"file {file}";
+                try
+                {
+                    bytes = File.ReadAllBytes(file);
+                }
+                catch (Exception)
+                {
+                    throw new CommandExitedException(
+                        $"Failed to read the file {file}; it may not exist nor be readable.",
+                        -1
+                    );
+                }
+            }
+
+            Transaction<NullAction> tx;
+            try
+            {
+                tx = Transaction<NullAction>.Deserialize(bytes, validate: false);
+            }
+            catch (Exception e)
+            {
+                throw new CommandExitedException(
+                    $"Failed to deserialize the {sourceName}: {e}",
+                    -1
+                );
+            }
+
+            if (!unsigned && !tx.Signature.Any())
+            {
+                throw new CommandExitedException(
+                    "The transaction is unsigned.  If you want to analyze it anyway, " +
+                        "use the -u/--unsigned option.",
+                    -1
+                );
+            }
+            else if (unsigned && tx.Signature.Any())
+            {
+                throw new CommandExitedException(
+                    "The transaction is signed, but the -u/--unsigned option is given.",
+                    -1
+                );
+            }
+
+            bool invalid = false;
+            if (!unsigned)
+            {
+                try
+                {
+                    tx.Validate();
+                }
+                catch (Exception e)
+                {
+                    invalid = true;
+                    Console.Error.WriteLine(
+                        "The signature from the transaction {0} is invalid: {1}",
+                        sourceName,
+                        e
+                    );
+                }
+
+                if (!invalid)
+                {
+                    Console.Error.WriteLine(
+                        "The signature from the transaction {0} is valid.",
+                        sourceName
+                    );
+                    invalid = !ignoreSignature;
```

Comment:
Wouldn't this line make the command return -1 when the signature is valid and ignoreSignature is false?

Reply:
Amended: 34bb8b4e39425f74ef0b4cc2fdda99dae23073e7.